### PR TITLE
CI: Use cached QEMU

### LINKED ISSUE #3119: CI: cache qemu, kernel, image for 2.x
The ci install scripts for 1.x look for built and cached qemu, kernel and image files in http://jenkins.katacontainers.io. In case of cache miss those artifacts are built on job time. 

For 2.x it doesn't have the jobs on jenkins to build and cache those artifacts, thus they are built for every CI job. This issue is about enabling such as caching mechanism to 2.x as well.

### PR DESCRIPTION
**Pull request by wainersm**
his changed the installer script to fetch and install a built QEMU
if a suitable version is found on the project's Jenkins.

To detect changes that would flag a cache mismatch it is employed
a simple algorithm to generate a SHA-256 from the files and scripts
used to build QEMU. The cached and current sha-256 is compared to
determine whether or not use the cached asset.

Fixes #3119
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

### CODE REVIEW

**Comment by wainersm** [comment]
@GabyCT @jcvenegas @chavafg hello!

This is just a draft but I would like to hear from you. Unlike kernel, we don't have a configuration number to check if the current and cached asset version match. As @jcvenegas suggested on #3489 , here I "implement" the idea of hashing all the files which are used to build qemu to compound a 'version sha' string which could be used for comparison.

File: .ci/lib.sh [hunk 1]
```diff
@@ -417,3 +417,24 @@ get_pr_changed_file_details()
...
+}
+
+gen_qemu_files_sha256sum() {
+	local pkg_dir="${kata_repo_dir}/tools/packaging"
+	# Files and dirs
+	files="${pkg_dir}/qemu \
+		${pkg_dir}/static-build/qemu \
+		${pkg_dir}/static-build/qemu/qemu.blacklist \
+		${pkg_dir}/static-build/scripts \
+		${pkg_dir}/scripts"
```
Comment (fidencio) [inline .ci/lib.sh:435]: I'd also add the QEMU version, coming from the version file, to the dance.

Right now we have patches which would accurately get any new change, but if we ever happen to use vanilla QEMU, not having the version could mislead us.

Comment (wainersm) [reply .ci/lib.sh:435]: Good catch @fidencio ! I will add that file too.

File: .ci/lib.sh [hunk 2]
```diff
@@ -417,3 +417,24 @@ get_pr_changed_file_details()
...
+sha256sum_from_files() {
+	# TODO: implement me!
+	echo "zzzzzzzzzzzzzzzzz"
+
+}
+
+gen_qemu_files_sha256sum() {
+	local pkg_dir="${kata_repo_dir}/tools/packaging"
+	# Files and dirs
+	files="${pkg_dir}/qemu \
```
Comment (jcvenegas) [inline .ci/lib.sh:431]: I think some of the scripts we use do `source` to other  scripts , even if we can not control what is sourced, it is still  a better approach than a number (that we need to modify manually) and I like the simplicity of the initial idea.

Comment (wainersm) [reply .ci/lib.sh:431]: Thanks for the feedback @jcvenegas . Still today I plan to send a working version of this PR.

**Comment by wainersm** [comment]
Updated this PR with a version which is good for reviews.

I will trigger some jobs that should build QEMU locally given that a suitable cached assert does not exist yet. Then later I will create the QEMU cahe job to check it can properly pull from there.

**Review by GabyCT — APPROVED** [review]

**Comment by wainersm** [comment]
@fidencio @GabyCT @jcvenegas , hi folks,

I created the job on Jenkins (http://jenkins.katacontainers.io/job/kata-containers-2.0-qemu-x86_64) which is already caching QEMU. The job's script pulls this PR, so it will needed to be adjusted afterwards.

I still have some improvements to be done on the algorithm that calculate the sha-256 to make it more reliable. But I will send it in another PR later.

It seems the mechanism is working fine. We can merge it, unless you find bugs or want me to implement more changes.

**Comment by GabyCT** [comment]
> @fidencio @GabyCT @jcvenegas , hi folks,
> 
> I created the job on Jenkins (http://jenkins.katacontainers.io/job/kata-containers-2.0-qemu-x86_64) which is already caching QEMU. The job's script pulls this PR, so it will needed to be adjusted afterwards.
> 
> I still have some improvements to be done on the algorithm that calculate the sha-256 to make it more reliable. But I will send it in another PR later.
> 
> It seems the mechanism is working fine. We can merge it, unless you find bugs or want me to implement more changes.

thank you @wainersm

**Review by devimc — APPROVED** [review]
thanks @wainersm

**Review by fidencio — APPROVED** [review]
lgtm, thanks @wainersm!

**Pull request merged**